### PR TITLE
Dafor improvements

### PR DESCRIPTION
**Pull request by kyrsjo**
Improvements to DAFOR:
* Allow command line arguments for input- and output files
* Fix all compilation warnings
* Trap some use of undefined variables.
* Stop with a non-zero return value in case of error
* Close all files when done.

### CODE REVIEW

**Comment by kyrsjo** [comment]
Nagfor is now mostly happy:
```
$ nagfor -c -O -abi=32 -dusty -maxcontin=60 -ieee=full -g90 dafor.f
Warning: Value 60 out of range for -maxcontin= option
NAG Fortran Compiler Release 6.0(Hibiya) Build 1064
Questionable: dafor.f, line 2137: Variable LC set but never referenced
Obsolescent: dafor.f, line 3: Fixed source form
Obsolescent: dafor.f, line 580: DO 10 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 673: DO 50 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 987: DO 1 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 1024: DO 5 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 1249: DO 123 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 1350: DO 171 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 1496: DO 320 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 1757: DO 1 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 1820: DO 10 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 1842: DO 20 ends neither with CONTINUE nor ENDDO
Obsolescent: dafor.f, line 1859: DO 30 ends neither with CONTINUE nor ENDDO
[NAG Fortran Compiler normal termination, 13 warnings]
```

**Comment by kyrsjo** [comment]
Ifort has infinite warnings about implicit types. Oh well.
```
fort -c -O -m32 dafor.f -warn all
dafor.f(11): warning #6717: This name has not been given an explicit type.   [PIENI]
      PARAMETER(PIENI=1D-17)
----------------^
dafor.f(12): warning #6717: This name has not been given an explicit type.   [ZERO]
      PARAMETER(ZERO=0.0D0,HALF=5.0D-1,ONE=1.0D0)
----------------^
```
etc. for 2136 lines:
```
$ ifort -c -O -m32 dafor.f -warn all 2>&1 | wc
   2136   10901  121465
```

James / Riccardo - can one of you review and merge? I think this will be great for the cmake building. Compatibility with "the old way", i.e. always use fort.1 and fort.2 is of course preserved. I removed fort.3 since it is anyway not used...

**Comment by kyrsjo** [comment]
OK, I checked the genereated .f files for the standard version (w/ gfortran) and DA version (w/ ifort). They are identical.

James, once you have read through the changes, unless there are comments please go ahead and merge.

File: SixTrack/dafor.f [hunk 1]
```diff
@@ -58,15 +63,64 @@ PROGRAM FOXY
       WRITE(6,*)' '
       WRITE(6,*)' '
 *
-   5  CONTINUE
-      open(1,file='fort.1',form='formatted',status='unknown')
-      open(2,file='fort.2',form='formatted',status='unknown')
-      open(3,file='fort.3',form='formatted',status='unknown')
+! 5    CONTINUE
+
```
Comment (JamesMolson) [inline SixTrack/dafor.f]: Whitespace at the end of this line.

File: SixTrack/dafor.f [hunk 2]
```diff
@@ -58,15 +63,64 @@ PROGRAM FOXY
...
 *
-   5  CONTINUE
-      open(1,file='fort.1',form='formatted',status='unknown')
-      open(2,file='fort.2',form='formatted',status='unknown')
-      open(3,file='fort.3',form='formatted',status='unknown')
+! 5    CONTINUE
+      
+      ! Read command line arguments -- inspired by read90
+      cmdarg_i = 0
+      do
```
Comment (JamesMolson) [inline SixTrack/dafor.f]: Whitespace at the end of this line.

File: SixTrack/dafor.f [hunk 3]
```diff
@@ -58,15 +63,64 @@ PROGRAM FOXY
...
+            fname_out = trim(cmdarg_arg)
+         else
+            write(6,*) "Error: Expected either no arguments OR "
+            write(6,*) "two arguements (input and output file name)"
+            stop 1
+         end if
+
+         cmdarg_i = cmdarg_i+1
+      end do
+
```
Comment (JamesMolson) [inline SixTrack/dafor.f]: here

File: SixTrack/dafor.f [hunk 4]
```diff
@@ -58,15 +63,64 @@ PROGRAM FOXY
...
+         write(6,*) "Error: Expected either no arguments OR "
+         write(6,*) "two arguements (input and output file name)"
+         stop 2
+      end if
+      if (fname_in.eq.fname_out) then
+         write(6,*)
+     &        "Error: input and output filename should not be the same!"
+         stop 3
+      end if
+
```
Comment (JamesMolson) [inline SixTrack/dafor.f]: here

File: SixTrack/dafor.f [hunk 5]
```diff
@@ -58,15 +63,64 @@ PROGRAM FOXY
...
+      open(1,file=fname_in,form='formatted',status='old')
+      open(2,file=fname_out,form='formatted',status='unknown') !status unknown -> allow overwrites
 *
       CALL predata
 *
       CALL PRECOM
 *
+      close(1)
+      close(2)
+
```
Comment (JamesMolson) [inline SixTrack/dafor.f]: here

File: SixTrack/dafor.f [hunk 6]
```diff
@@ -2122,14 +2196,16 @@ SUBROUTINE VNAM(IA, IN, A,NA,LA, ITA, ISCRTY)
      *            '81','82','83','84','85','86','87','88','89','90',
      *            '91','92','93','94','95','96','97','98','99'/
 *
+      ND = -4242
+
```
Comment (JamesMolson) [inline SixTrack/dafor.f]: here

File: SixTrack/dafor.f [hunk 7]
```diff
@@ -2162,6 +2238,12 @@ SUBROUTINE VNAM(IA, IN, A,NA,LA, ITA, ISCRTY)
          WRITE(6,'(1X,A)') '!!! ERROR, IA = 0 IN VNAM'
          WRITE(2,'(1X,A)') '!!! ERROR, IA = 0 IN VNAM'
       ENDIF
+
+      if (ND .eq. -4242) then
+         write(6,*) "!!! ERROR, ND was not initialized in VNAM"
+         stop 15
+      end if
+
```
Comment (JamesMolson) [inline SixTrack/dafor.f]: and here

File: SixTrack/dafor.f (hunk 1 above)
Comment (kyrsjo) [inline SixTrack/dafor.f]: This one is just in line with the general indent level. -> keep.

**Pull request merged**